### PR TITLE
Allow multiple B backends

### PR DESCRIPTION
**Pull request by jindrichskupa**
We have multiple development/testing backends, so I added support for multiple B systems:

* allow using `-b` param multiple times
* update usage

Thanks for your work.

### CODE REVIEW

File: teeproxy.go [hunk 1]
```diff
@@ -92,54 +105,60 @@ func SchemeAndHost(url string) (scheme, hostname string) {
...
 	if *percent == 100.0 || h.Randomizer.Float64()*100 < *percent {
-		alternativeRequest, productionRequest = DuplicateRequest(req)
-		go func() {
-			defer func() {
-				if r := recover(); r != nil && *debug {
-					log.Println("Recovered in ServeHTTP(alternate request) from:", r)
-				}
-			}()
+		for _, alt := range h.Alternatives {
+			alternativeRequest = DuplicateRequest(req)
```
Comment (chrislusf) [inline teeproxy.go]: Does this replicated request work for HTTP POST requests?

I guess the second time the HTTP POST body will be empty.

Comment (jindrichskupa) [reply teeproxy.go]: Sure, the problem should be fixed. Thanks for the notice.

Comment (jindrichskupa) [reply teeproxy.go]: Fixed here: https://github.com/chrislusf/teeproxy/pull/47/commits/a082301a238ce4b6655a036eeddff2b62d8f7e08

Comment (chrislusf) [reply teeproxy.go]: The replicated request is in memory, but the original request body could have been consumed.

Comment (jindrichskupa) [reply teeproxy.go]: I am confused about how to repair it. Give me some hint to do it better.

Comment (chrislusf) [reply teeproxy.go]: The "req" body will be consumed on the first DuplicateRequest(req) execution, and the second alternativeRequest can only get an empty body.

**Comment by chrislusf** [comment]
Thanks! I read it wrong last time.

How about optimize a bit more? If the request.body is already NopCloser, just reuse it instead of creating a new byte buffer?

**Pull request merged**